tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,352 @@
+#NAT BASED NETWORK
+##Prepare Test
+my $vnet_nated = <<'EOF';
+<network>
+  <name>vnet_nated</name>
+  <bridge name="virbr0"/>
+  <forward mode="nat"/>
+  <ip address="192.168.122.1" netmask="255.255.255.0">
+    <dhcp>
+      <range start="192.168.122.2" end="192.168.122.254"/>
+    </dhcp>
+  </ip>
+</network>
+EOF
+    die "There was not the default(NAT BASED NETWORK) virtual network" if (script_run('virsh net-list --all | grep default') != 0);
+
+    #Create NAT BASED NETWORK
+    assert_script_run("virsh net-info default");
+    assert_script_run("virsh net-dumpxml default |tee libvirt_default.xml");
+    assert_script_run("virsh net-undefine default");
+    assert_script_run("echo '$vnet_nated' >> vnet_nated.xml");
+    assert_script_run("virsh net-create vnet_nated.xml");
+    save_screenshot;
+    upload_logs "vnet_nated.xml";
+    assert_script_run("rm -rf vnet_nated.xml");
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "NAT BASED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_nated --live",60);
+        my $gi_vnet_nated = script_output("sleep 60; journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        #Test NAT BASED NETWORK
+        assert_script_run("ssh root\@$gi_vnet_nated 'rcnetwork restart'", 60);
+        assert_script_run("ssh root\@$gi_vnet_nated 'ping -c2 -W1 openqa.suse.de'",60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy NATed Network
+    assert_script_run("virsh net-destroy vnet_nated");
+    save_screenshot;
+
+    #Restore default(NATed Network)
+    my $default_path  = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0){assert_script_run("virsh net-define $default_path",60);assert_script_run("rm -rf $default_path");}
+
+##ROUTED NETWORK
+##Prepare Test
+my $vnet_routed = <<'EOF';
+<network>
+ <name>vnet_routed</name>
+ <bridge name="virbr1"/>
+ <forward mode="route"/>
+ <ip address="192.168.129.1" netmask="255.255.255.0">                                                   <dhcp>
+      <range start="192.168.129.2" end="192.168.129.254" />                                             </dhcp>
+ </ip>
+</network>
+EOF
+
+    #Create and Start ROUTED NETWORK
+    assert_script_run("echo '$vnet_routed' >> vnet_routed.xml");
+    assert_script_run("virsh net-create vnet_routed.xml");
+    save_screenshot;
+    upload_logs "vnet_routed.xml";
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "ROUTED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_routed --live");
+        my $gi_vnet_routed = script_output("sleep 30;journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        #Confirm ROUTED NETWORK
+        #TODO 
+	#There should be added the static routes to LAN router, but we have not permission 
```

Comment:
What does it mean that we don't have permission?

Reply:
For Libvirt ROUTE NETWORK, it need to modify our LAB router to add the static route setting, as I known. But, I do not have permission to modify our LAB router, here just only IT department have this permission to modify LAB router.

Reply:
Great idea, I think, there will be create two private router on one VM Server, one router will help to ping from guest(deployed guest from the other router) to external. So, as the TODO mentioned, before confirm the libvirt route network setting, we will keep the current route network setting and WITHOUT testing ping from guest to external for now. 
Moreover, after confirm libvirt route network setting form developer or network testing team, will submit new PR to update existed route network codes.

Reply:
No, my point is, that if you create two guests on the hypervisor and each guest will be connected to different internal network, you can then route the traffic between those two. Or even better you can have one guest connected to a routed network and the other guest can be connected to a bridge. In both scenarios the host will act as a router so you do not have to modify any external device.

Reply:
Great point, regarding ROUTE NETWORK, I think, please updated your comments to bsc#1149778 - [KVM] Guest system do not go out via ROUTED NETWORK on KVM Host Server, I had filed.

Reply:
In my opinion there is no bug, I just suggested you a solution how to test the routed network without modifying any external router. Once again:
> You can have two guests in two different routed networks so then the host can route their traffic.

This can be implemented here.

Reply:
Refer to your suggestion, had implemented that  two guests in two different routed networks so then the host can route their traffic.